faunadb-java/src/main/java/com/faunadb/client/query/Pagination.java

```diff
@@ -0,0 +1,175 @@
+package com.faunadb.client.query;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.faunadb.client.types.Value;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableMap;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A pagination expression.
+ * <p>
+ * <b>Example:</b>
+ * We can paginate all instances of a class, with its evets, in pages with 4 elements only.
+ * <pre>{@code
+ * client.query(
+ *    Paginate(Match(Ref("indexes/all_spells"))
+ *      .withEvents(true)
+ *      .withSize(4)
+ * );
+ * }</pre>
+ *
+ * @see Language#Paginate(Expr)
+ * @see <a href="https://faunadb.com/documentation/queries#read_functions">FaunaDB Read Functions</a>
+ */
+public final class Pagination extends Fn.Call {
```

Comment:
That's tricky. 

`Fn.Call` is a `Value` that serialises itself as a JS Object. Similar to `ObjectV` but, `ObjectV` serialises itself with "object" key as its root key. I was trying to delegate escaping logic to each one of the `Expr` subclasses. 

If we make pagination an `Expr` directly, it will not compose with `ArrayV` and `ObjectV` as they expect to be constructed with instances of `Value` only.

https://github.com/faunadb/faunadb-jvm/blob/jvm_syntax/faunadb-java/src/main/java/com/faunadb/client/query/Expr.java#L34

Is that a problem?